### DISCUSSION TITLE
The limit of '0' for Top query has been exceeded error when setting a 'top' value in swagger

### DISCUSSION THREAD

**Discussion by Christina005** [dotnet/aspnet-api-versioning#965] [General]
I have tested the sample in your solution 'ODataOpenApiExample', when setting a  'top' value, version 2 get people for example, options.Validate( validationSettings ) throws an ODataException: The limit of '0' for Top query has been exceeded. The value from the incoming request is '1'

Is there a fix for this? I must resolve this issue in my own project which has the same problem

**Comment by commonsensesoftware** [comment]
#944 has a bit more of a succinct overview of the issue, but the problem is ultimately in OData. I've been asked this question a lot so I provided a comprehensive analysis in the OData/AspNetCoreOData#695 bug. There are some workarounds, but the real fix is in the hands of the OData team.
